astropy/coordinates/sky_coordinate.py

```diff
@@ -1359,11 +1360,29 @@ def radial_velocity_correction(self, kind='barycentric', obstime=None,
                              "'{}'".format(kind))
 
         gcrs_p, gcrs_v = location.get_gcrs_posvel(obstime)
-        gtarg = self.transform_to(GCRS(obstime=obstime,
-                                       obsgeoloc=gcrs_p,
-                                       obsgeovel=gcrs_v))
-        targcart = gtarg.represent_as(UnitSphericalRepresentation).to_cartesian()
-        return targcart.dot(v_origin_to_earth + gcrs_v)
+        # transforming to GCRS is not the correct thing to do here, since we don't want to
+        # include aberration (or light deflection)? Instead, only apply parallax if necessary
+        if self.data.__class__ is UnitSphericalRepresentation:
+            targcart = self.represent_as(UnitSphericalRepresentation).to_cartesian()
+        else:
+            # skycoord has distances so apply parallax
+            obs_icrs_cart = get_body_barycentric('earth', obstime) + gcrs_p
+            icrs_cart = self.cartesian
+            targcart = (icrs_cart - obs_icrs_cart).represent_as(UnitSphericalRepresentation).to_cartesian()
```

Comment:
This is a bit slow, since you have to do all the needless trig. I'd do:
```
targcart = icrs_cart - obs_icrs_cart
targcart /= targcart.norm()
```

Reply:
Much better.